### PR TITLE
Improve code readability. Exhaustive changes for each class of cleanup.

### PR DESCRIPTION
**Pull request by practicalswift**
Improve code readability:
* Use `auto*` instead of `auto` for pointers
* ~~Avoid shadowing variable~~
* Separate assignment from function call
* ~~Remove redundant code~~

These changes are all exhaustive.

### CODE REVIEW

File: src/qt/askpassphrasedialog.cpp [hunk 1]
```diff
@@ -238,10 +238,10 @@ bool AskPassphraseDialog::event(QEvent *event)
...
 {
     ui->toggleShowPasswordButton->setDown(show);
-    const auto mode = show ? QLineEdit::Normal : QLineEdit::Password;
-    ui->passEdit1->setEchoMode(mode);
-    ui->passEdit2->setEchoMode(mode);
-    ui->passEdit3->setEchoMode(mode);
+    const auto echoMode = show ? QLineEdit::Normal : QLineEdit::Password;
+    ui->passEdit1->setEchoMode(echoMode);
+    ui->passEdit2->setEchoMode(echoMode);
+    ui->passEdit3->setEchoMode(echoMode);
```
Comment (MarcoFalke) [inline src/qt/askpassphrasedialog.cpp]: We don't do these changes, since it is non-trivial to prove the refactoring does the right thing:  #10136

Comment (practicalswift) [reply src/qt/askpassphrasedialog.cpp]: Wasn't the problem that people used ancient compilers?

This was one of the comments:

> And there is still value in compiling with Wshadow periodically (with a carefully chosen compiler) and seeing if it finds any bug.

I'm using a carefully chosen compiler :-)

**Comment by practicalswift** [comment]
@MarcoFalke I've removed what I see as the most "subjective" changes. Do you think that any of the remaining ones are acceptable or should I close? :-)

File: src/qt/addressbookpage.cpp [hunk 2]
```diff
@@ -38,7 +38,7 @@ class AddressBookSortFilterProxyModel final : public QSortFilterProxyModel
 protected:
     bool filterAcceptsRow(int row, const QModelIndex& parent) const
     {
-        auto model = sourceModel();
+        auto* model = sourceModel();
```
Comment (MarcoFalke) [inline src/qt/addressbookpage.cpp]: This change makes it harder to switch to smart pointers later, no?

File: src/qt/askpassphrasedialog.cpp (hunk 1 above)
Comment (practicalswift) [reply src/qt/askpassphrasedialog.cpp]: Oh, I misread a report. Sorry. Reverted the shadowing change :-)

**Comment by Diving-Or-Dove-In** [comment]
is the english oxford dictionary?

**Pull request closed without merging**